Doc/c-api/tuple.rst

```diff
@@ -35,6 +35,10 @@ Tuple Objects
 
    Return a new tuple object of size *len*, or *NULL* on failure.
 
+   .. versionchanged:: 3.7
+
+      ``PyTuple_New(0)`` always succeeds and never returns *NULL*.
```

Comment:
Maybe comments around `PyTuple_New` enough if we just don't want to forget about it?